unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -138,6 +138,15 @@ public int numChars() {
   }
 
   /**
+   * Returns a 64-bit integer that can be used as the prefix used in sorting.
+   */
+  public long getPrefix() {
+    long p = PlatformDependent.UNSAFE.getLong(base, offset);
+    p = java.lang.Long.reverseBytes(p);
```

Comment:
We should check the numBytes here, or get random bytes if it's less than 8 bytes.